### PR TITLE
[spirv] Translate casting to bool (vector) values

### PR DESCRIPTION
**Pull request by antiagainst**
* Supported converting scalar/vector sint/uint/float values into
  scalar/vector bool values.

### CODE REVIEW

File: tools/clang/lib/SPIRV/EmitSPIRVAction.cpp [hunk 1]
```diff
@@ -1089,6 +1153,38 @@ case BO_##kind : {                                                             \
...
+      if (size == 1)
+        return elemZeroId;
+
+      llvm::SmallVector<uint32_t, 4> elements(size, elemZeroId);
+
+      const uint32_t vecTypeId = typeTranslator.translateType(type);
+      return theBuilder.getConstantComposite(vecTypeId, elements);
+    }
+
+    emitError("getting value 0 for type '%0' unimplemented") << type;
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/EmitSPIRVAction.cpp]: s/type/type.getAsString()

File: tools/clang/test/CodeGenSPIRV/var.init.hlsl [hunk 2]
```diff
@@ -52,6 +58,13 @@ float4 main(float component: COLOR) : SV_TARGET {
 // CHECK-NEXT: OpStore %o [[oinit]]
     float4 o = float4(1.0, j, 3.0, j);      // Mixed case
 
+    int2 p = {1, 2}; // All components are constants
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/var.init.hlsl]: where are the CHECKs for `p` ?

Comment (antiagainst) [inline tools/clang/test/CodeGenSPIRV/var.init.hlsl]: At line 29 in the above. Initializer is used for this case instead of `OpStore`.

Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/var.init.hlsl]: Ah! cool. 👍

File: tools/clang/lib/SPIRV/EmitSPIRVAction.cpp (hunk 1 above)
Comment (antiagainst) [inline tools/clang/lib/SPIRV/EmitSPIRVAction.cpp]: Done

**Comment by antiagainst** [comment]
Hmm, the most recent two failures were caused by exceeding the 1h limit for tests. Not sure why it suddenly taking more than ~15min to run the tests now.

**Comment by ehsannas** [comment]
hmm.. That's odd...

**Comment by antiagainst** [comment]
I'm cleaned up the history and force pushed a new commit into this branch to see if it works.

**Comment by antiagainst** [comment]
Okay, Appveyor is happy this time.

**Pull request merged**